Houseclub/src/main/java/me/grishka/houseclub/fragments/InviteListFragment.java

```diff
@@ -0,0 +1,273 @@
+
+public class InviteListFragment extends SearchListFragment {
+
+    private InviteListAdapter adapter;
+
+    private static final int REQUEST_READ_CONTACTS = 0;
+
+    public InviteListFragment() {
+        min_query_lenght = 0;
+    }
+
+    @Override
+    protected RecyclerView.Adapter getAdapter(){
+        if(adapter==null){
+            adapter=new InviteListAdapter();
+        }
+        return adapter;
+    }
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions,
+                                           @NonNull int[] grantResults) {
+        if (requestCode == REQUEST_READ_CONTACTS) {
+            if (grantResults.length == 1 && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+                getContactList("");
+            }
+        }
+    }
+
+    private boolean askContactsPermission() {
+
+        if (getContext().checkSelfPermission(READ_CONTACTS) == PackageManager.PERMISSION_GRANTED) {
+            return true;
+        }
+        requestPermissions(new String[]{READ_CONTACTS}, REQUEST_READ_CONTACTS);
+        return false;
+    }
+
+    private void readContacts(String query) {
+        if (!askContactsPermission()) {
+            return;
+        } else {
+            getContactList(query);
+        }
+    }
+
+    private void getContactList(String query) {
+
+        List<FullUser> users = new ArrayList<>();
+        int count = 0, limit = 10;
+
+        ContentResolver cr = getContext().getContentResolver();
+        Cursor cur = cr.query(ContactsContract.Contacts.CONTENT_URI,
+                null, null, null, null);
+
+        if ((cur != null ? cur.getCount() : 0) > 0) {
+            while (cur != null && cur.moveToNext()) {
+
+                String id = cur.getString(
+                        cur.getColumnIndex(ContactsContract.Contacts._ID));
+
+                String name = cur.getString(cur.getColumnIndex(
+                        ContactsContract.Contacts.DISPLAY_NAME));
+
+                if (cur.getInt(cur.getColumnIndex(
+                        ContactsContract.Contacts.HAS_PHONE_NUMBER)) > 0) {
+                    Cursor pCur = cr.query(
+                            ContactsContract.CommonDataKinds.Phone.CONTENT_URI,
+                            null,
+                            ContactsContract.CommonDataKinds.Phone.CONTACT_ID + " = ?",
+                            new String[]{id}, null);
+                    while (pCur.moveToNext()) {
+                        String phoneNo = pCur.getString(pCur.getColumnIndex(
+                                ContactsContract.CommonDataKinds.Phone.NUMBER));
+
+                        if (query.equals("") || (
+                                Pattern.compile(Pattern.quote(query), Pattern.CASE_INSENSITIVE).matcher(name).find() ||
+                                        Pattern.compile(Pattern.quote(query), Pattern.CASE_INSENSITIVE).matcher(phoneNo).find())) {
```

Comment:
compiled 2nd time

Reply:
phoneNo

Reply:
Is there anyone who  can send me invitation